### ISSUE TITLE
Pull EBI study into Qiita

### ISSUE THREAD

**Issue by sjanssen2** [biocore/qiita#2322]
Our lab's strength is do easily do meta analysis of existing studies. Currently, we can do that without much effort with all studies in Qiita. Sadly, not everyone is using Qiita. Thus, there are exciting studies whose sequence files are publicly available in EBI but not in Qiita.
It would be awesome, if there exists a way to pull those EBI projects into Qiita as a new study (sequence data + metadata) to extend the pool of samples for metadata analysis.

I am asking, because Sarkis' lab is asking to do some meta analysis of existing PD studies and at least two are in EBI but not in Qiita. Thinking a little bit ahead would mean to compare not only PD but also other auto immune diseases. Thus, it is very likely that we want to add in even more projects.

**Comment by antgonza** [comment]
👍  

My main question is how to actually do this. Some tentative steps:
1. User creates study
2. User can add Study EBI accession (should this be the only point of entry?)
3. A qiita plugin (or should this be part of Qiita?) validates and retrieves all samples/runs and sorts them by prep (experiment accession) and adds everything to Qiita. At the end you create a study with a sample and (possibly) multiple preps and all the runs as sandboxed. Obviously, accessions are added to the study so it's shown as submitted.
4. The user can ask an admin to make public.

**Comment by josenavas** [comment]
I think this is a good idea, I also like the general implementation structure that @antgonza proposes. I think the entry point should be different: instead of the user creating the study, add a new page that is "Import study" and the user provides the Study EBI accession number. The title and most of the other information is already available in EBI, so we should be able to populate those entries.

In addition, when the accession number is provided, the system should check if that EBI accession number is already present in Qiita - this way we avoid duplicated studies (for example, if the study is still sandboxed and needs to be made public).

Finally, I think the code can live in Qiita (as the current code for EBI submission lives in Qiita), and use the internal plugin system (i.e. Qiita as a Qiita plugin :P).

**Comment by adswafford** [comment]
Daniel and I had a conversation about this issue last week and early this week where we had an alternative idea which was to start pulling in all the 16S V4 studies from EBI (that we didn't submit already ourselves) and making them public so we don't have data duplication or rely on individual users to pull them in manually. If we want to semi-automate things and let user-desire drive the prioritization of which studies to pull in then I think the implementation described above is fine, but if we instead want to just set things up to pull it all in then this could save us the time of making a UI for users to do it.

**Comment by jdereus** [comment]
And what is the volume of data that you would anticipate from a pull of all 16S V4 studies?

**Comment by antgonza** [comment]
Right, that's another option. Some things to consider about this approach is (1) storage, (2) how often do you "crawl" EBI for new studies?, and (3) should all be downloaded or should we have some kind of prioritization (public in EBI doesn't mean published or nice to have)?

**Comment by wasade** [comment]
To add, not retaining the ebi sequence data locally for those fetches, just
process and keep summarized results (e.g. biom)

**Comment by wasade** [comment]
So storage is negligible

**Comment by adswafford** [comment]
If we just keep the biom files though we can't guarantee consistent processing. However, this brings up a related issue: once sequences are processed, should we have a way for users to delete the original FASTQ files since they're not needed? I can see the rationale for hanging onto the demuxed files but not the original.

**Comment by jdereus** [comment]
But there would still be an associated spike in storage at some point.

What does temp storage look like?
Is only single study pulled down, processed, samples "discarded", move to next one?

**Comment by jdereus** [comment]
Is there the situation where someone else comes along, views public study and wants to reprocess with the study original fastq files?

**Comment by wasade** [comment]
If we perform the processing of the EBI data, then we can assert consistent
processing and associate the correct processing parameters. Performing
additional processing for a (pure) EBI study would require fetching the raw
sequence data again, but that's reasonably fast.

**Comment by sjanssen2** [comment]
I tend to prefer @josenavas suggestion for the entry point.
One more feature request: would it be possible to automatically pull the according publication and fill the study description in Qiita with the paper's abstract?

**Comment by antgonza** [comment]
@sjanssen2, sure ...

**Comment by antgonza** [comment]
@wasade, that's an excellent point: we should be able to delete all EBI submitted files (free space) and just download at the time of processing ... worth another issue so we can discuss?

**Comment by antgonza** [comment]
Just thought about other thing to consider: A public study in EBI doesn't necessarily needs to be MIxS compatible, what should we add with missing columns? Currently: `XXQIITAXX`

**Comment by antgonza** [comment]
Merging 2 issues
-----------------

Moving comments from https://github.com/biocore/qiita/issues/2826 here. 

EBI Import codes currently execute in Tornado's process space, and need to be moved to a Qiita plugin.

- [ ] Have EBI Import artifact automatically call Trim, Deblur. A good way to know if EBI import failed is if Trim fails. If the new artifact can create a Trim processing job, etc. that would be beneficial.

- [ ] Framework for EBI import needed in Qiita self-plugin. Charlie will create an empty framework to support EBI import as a Qiita self-plugin.

- [ ] Enhance documentation for installation, configuration, and security of plugins. Enhance documentation to describe the contents of plugin configuration files, where they should be placed, how keys can be generated, how to create whatever python environment is needed, etc.

- [ ] Test cases for EBI import. Test cases for EBI import as it is currently developed. We will migrate these as we migrate EBI import from code living in the GUI to a Qiita plugin.

Will assign Den as a co-assignee, once he is added as a contributor.
